### ISSUE TITLE
how to separate an duet audio?

### ISSUE THREAD

**Issue by MMMMichaelzhang** [yoyololicon/duet-svs-diffusion#2]
how to separate an duet audio?

**Comment by yoyolicoris** [comment]
Hi,
if you were asking for the algorithm details, you can check the function here
https://github.com/yoyololicon/duet-svs-diffusion/blob/69c8276ee91f030aa6686c3a5e18c2348737119c/eval.py#L19
We didn't have a simple script that could be used immediately for simple use cases, but you can start something from there.
I hope this helps.

**Comment by MMMMichaelzhang** [comment]
Please make a simple script for separation, so we can directly test the result.Thanks.

**Comment by MMMMichaelzhang** [comment]
pred = separate_mixture(
                                sub_m,
                                noise,
                                denoise_fn,
                                sigmas,
                                s_churn=s_churn,
                                cond=cond,
                                cond_index=overlap_size,
                                use_tqdm=False,
                                gaussian=False,
                            )
in 'eval.py' noise and overlap_size is related to the source file, if I only have mix files,how to set 'noise' and 'overlap_size'?

**Comment by MMMMichaelzhang** [comment]
mixture_path = 'input.wav'
mixture, sr = librosa.load(mixture_path, sr=44100)
mixture = torch.as_tensor(mixture, dtype=torch.float32)

model_config_path = 'model/config.yaml'
model_path = 'model/last.ckpt'
T = 100 #Number of diffusion steps
s_churn = 40.0 #S churn
hop_length = 1024

config_path, config_name = os.path.split(model_config_path)

with hydra.initialize(config_path=config_path):
    cfg = hydra.compose(config_name=config_name)

sr = cfg.sampling_rate
length = cfg.length

model = hydra.utils.instantiate(cfg.model)
vctk_checkpoint = torch.load(
    model_path,
    map_location="cpu",
)
model.load_state_dict(vctk_checkpoint["state_dict"])
diffusion_schedule = hydra.utils.instantiate(
    cfg.callbacks.audio_samples_logger.diffusion_schedule
)

model = model.cuda()
model.eval()
diffusion_schedule = diffusion_schedule.cuda()

inner_denoise_fn = model.model.diffusion.diffusion.denoise_fn
def denoise_fn(x, sigma):
    x = x.unsqueeze(1)
    return inner_denoise_fn(x, sigma=sigma).squeeze(1)

sigmas = diffusion_schedule(T, "cuda")

x = mixture.cuda()
original_length = x.numel()
padding = length - (original_length % length)
if padding < length:
    x = torch.cat([x, x.new_zeros(padding)], dim=0)

result = separate_mixture(
    x,
    torch.randn(2, x.numel()).cuda(),
    denoise_fn,
    sigmas,
    s_churn=s_churn,
    use_tqdm=False,
)[:, :original_length]
est = result.squeeze().cpu().numpy()
filename = os.path.splitext(os.path.basename(mixture_path))[0]
for i, s in enumerate(est):
    out_path = 'outputs' + '/' + f"{filename}_{i}.wav"
    sf.write(out_path, s, 44100, "PCM_16")


Is it correct?And the speed is very slow,what should i do?Thanks.

**Comment by MMMMichaelzhang** [comment]
When the mix file is separated into two files like A and B, there is B in A and A in B, and they are constantly interspersed. How to solve this problem? It seems that adjusting the s-churn can solve something?//Thanks

**Comment by MMMMichaelzhang** [comment]
When the mix file is separated into two files like A and B, there is B in A and A in B, and they are constantly interspersed. How to solve this problem? It seems that adjusting the s-churn can solve something?//Thanks

**Comment by MMMMichaelzhang** [comment]
[Uploading 3files.zip…]()
